.class public final Lkotlinx/coroutines/ExceptionsKt;
.super Ljava/lang/Object;
.source "Exceptions.kt"


# annotations
.annotation system Ldalvik/annotation/SourceDebugExtension;
    value = "SMAP\nExceptions.kt\nKotlin\n*S Kotlin\n*F\n+ 1 Exceptions.kt\nkotlinx/coroutines/ExceptionsKt\n+ 2 fake.kt\nkotlin/jvm/internal/FakeKt\n*L\n1#1,76:1\n1#2:77\n*E\n"
.end annotation

.annotation runtime Lkotlin/Metadata;
    d1 = {
        "\u0000 \n\u0000\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0000\n\u0002\u0010\u0003\n\u0000\n\u0002\u0010\u0002\n\u0002\u0008\u0002\u001a\u001e\u0010\u0000\u001a\u00060\u0001j\u0002`\u00022\u0008\u0010\u0003\u001a\u0004\u0018\u00010\u00042\u0008\u0010\u0005\u001a\u0004\u0018\u00010\u0006\u001a\u0015\u0010\u0007\u001a\u00020\u0008*\u00020\u00062\u0006\u0010\t\u001a\u00020\u0006H\u0080\u0008*\n\u0010\u0000\"\u00020\u00012\u00020\u0001\u00a8\u0006\n"
    }
    d2 = {
        "CancellationException",
        "Ljava/util/concurrent/CancellationException;",
        "Lkotlinx/coroutines/CancellationException;",
        "message",
        "",
        "cause",
        "",
        "addSuppressedThrowable",
        "",
        "other",
        "kotlinx-coroutines-core"
    }
    k = 0x2
    mv = {
        0x1,
        0x6,
        0x0
    }
    xi = 0x30
.end annotation


# direct methods
.method public static final CancellationException(Ljava/lang/String;Ljava/lang/Throwable;BFCS)V
    .locals 0

	goto/32 :l_VWjRwuPrKOQNRZiI_0

	nop

	:l_zHQYqyyrnxDjjLUm_6
    return-void

	:after_last_instruction

	goto/32 :l_eKLJGZJPVPavtOYK_7

	nop

	:l_eLlyvyTtLElaJitj_4
    add-int p3, p2, p1

	goto/32 :l_zzEzBqQgKhDmuVch_5

	nop

	:l_zzEzBqQgKhDmuVch_5
    int-to-double p0, p3

	goto/32 :l_zHQYqyyrnxDjjLUm_6

	nop

	:l_eKLJGZJPVPavtOYK_7
	goto/32 :before_first_instruction

	:l_sAXBADqlkDVuJvmX_2
    const/16 p1, 0xd2

	goto/32 :l_WGlRBEGyHniqLqFp_3

	nop

	:l_VWjRwuPrKOQNRZiI_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_ZdTlajvDvKjldhEc_1

	nop

	:l_WGlRBEGyHniqLqFp_3
    mul-int p2, p0, p1

	goto/32 :l_eLlyvyTtLElaJitj_4

	nop

	:l_ZdTlajvDvKjldhEc_1
    const/16 p0, 0x2a

	goto/32 :l_sAXBADqlkDVuJvmX_2

	nop

.end method

.method public static final CancellationException(Ljava/lang/String;Ljava/lang/Throwable;FCBS)V
    .locals 0

	goto/32 :l_NTjazDyjycPXUvcX_0

	nop

	:l_IevVsBqgiJhwLtfj_2
    const/16 p1, 0xd2

	goto/32 :l_dOEmuICGXTZEQhCZ_3

	nop

	:l_dOEmuICGXTZEQhCZ_3
    mul-int p2, p0, p1

	goto/32 :l_qLkOpNFTKrzrjKAZ_4

	nop

	:l_YRQuyRtshSHFxxfG_5
    int-to-double p0, p3

	goto/32 :l_ndfksJakRUrUAHhk_6

	nop

	:l_qLkOpNFTKrzrjKAZ_4
    add-int p3, p2, p1

	goto/32 :l_YRQuyRtshSHFxxfG_5

	nop

	:l_TAhLfwnlgzKYGEkg_7
	goto/32 :before_first_instruction

	:l_CXRahtjpctXNFaxd_1
    const/16 p0, 0x2a

	goto/32 :l_IevVsBqgiJhwLtfj_2

	nop

	:l_ndfksJakRUrUAHhk_6
    return-void

	:after_last_instruction

	goto/32 :l_TAhLfwnlgzKYGEkg_7

	nop

	:l_NTjazDyjycPXUvcX_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_CXRahtjpctXNFaxd_1

	nop

.end method

.method public static final CancellationException(Ljava/lang/String;Ljava/lang/Throwable;SCBF)V
    .locals 0

	goto/32 :l_xGEiJYPNcrorATJm_0

	nop

	:l_gEyneRVrvGZkDDmC_3
    mul-int p2, p0, p1

	goto/32 :l_RVMEslmFFbPCunhk_4

	nop

	:l_uRMewhEFmrUIxYXf_2
    const/16 p1, 0xd2

	goto/32 :l_gEyneRVrvGZkDDmC_3

	nop

	:l_vQPiIWsJSGRoBJKo_5
    int-to-double p0, p3

	goto/32 :l_aaXAcDrjMDPwEhyA_6

	nop

	:l_aaXAcDrjMDPwEhyA_6
    return-void

	:after_last_instruction

	goto/32 :l_FMUeuqWSzYxyHEHZ_7

	nop

	:l_XSTTonKnYkJolUsi_1
    const/16 p0, 0x2a

	goto/32 :l_uRMewhEFmrUIxYXf_2

	nop

	:l_xGEiJYPNcrorATJm_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_XSTTonKnYkJolUsi_1

	nop

	:l_RVMEslmFFbPCunhk_4
    add-int p3, p2, p1

	goto/32 :l_vQPiIWsJSGRoBJKo_5

	nop

	:l_FMUeuqWSzYxyHEHZ_7
	goto/32 :before_first_instruction

.end method

.method public static final CancellationException(Ljava/lang/String;Ljava/lang/Throwable;)Ljava/util/concurrent/CancellationException;
    .locals 3

	goto/32 :l_pLUdZnKRIwfiOrMd_0

	nop

	:l_sQpepWjlNhHmjAcu_3
	rem-int v0, v0, v1
	goto/32 :l_MfkqwSizNxZHcoIn_4

	nop

	:l_WupfgoSsDenuXLse_14
	goto/32 :JEkfJycygFjUDZTb
	:l_DvqoTqQboeefzWwr_12
    return-object v0

	:after_last_instruction

	goto/32 :l_FEVFGlmRBqihOwff_13

	nop

	:l_GAQqtamEJNQbZIBj_1
	const v1, 23
	goto/32 :l_eYvhGSfXImWbtKDn_2

	nop

	:l_eYvhGSfXImWbtKDn_2
	add-int v0, v0, v1
	goto/32 :l_sQpepWjlNhHmjAcu_3

	nop

	:l_SNUPdWhEvcAcEkoe_8
    invoke-direct {v0, p0}, Ljava/util/concurrent/CancellationException;-><init>(Ljava/lang/String;)V

	goto/32 :l_FFKacvnhSMJMuNOM_9

	nop

	:l_LjNjmbVBIcxpEdZo_5
	goto/32 :FopirqlXGSYNKUeb
	:RknTirAaGDBzUHvQ
	:JEkfJycygFjUDZTb

	goto/32 :l_LcfAIjovzlKiFwUS_6

	nop

	:l_NqxqxkYmJiNzPUSU_7
    new-instance v0, Ljava/util/concurrent/CancellationException;

	goto/32 :l_SNUPdWhEvcAcEkoe_8

	nop

	:l_pLUdZnKRIwfiOrMd_0
	const v0, 12
	goto/32 :l_GAQqtamEJNQbZIBj_1

	nop

	:l_geQizlSmDvYMWlMG_10
    const/4 v2, 0x0

    .line 22
    .local v2, "$i$a$-apply-ExceptionsKt$CancellationException$1":I
	goto/32 :l_qHnSDfSBHFzuzSZs_11

	nop

	:l_FEVFGlmRBqihOwff_13
	goto/32 :before_first_instruction

	:FopirqlXGSYNKUeb
	goto/32 :l_WupfgoSsDenuXLse_14

	nop

	:l_MfkqwSizNxZHcoIn_4
	if-lez v0, :gl_zfvHwEoRXhbMkYYA

	goto/32 :RknTirAaGDBzUHvQ

	:gl_zfvHwEoRXhbMkYYA	goto/32 :l_LjNjmbVBIcxpEdZo_5

	nop

	:l_LcfAIjovzlKiFwUS_6
	goto/32 :after_last_instruction

	:before_first_instruction
    .param p0, "message"    # Ljava/lang/String;
    .param p1, "cause"    # Ljava/lang/Throwable;

    .line 22
	goto/32 :l_NqxqxkYmJiNzPUSU_7

	nop

	:l_FFKacvnhSMJMuNOM_9
    move-object v1, v0

    .line 77
    .local v1, "$this$CancellationException_u24lambda_u2d0":Ljava/util/concurrent/CancellationException;
	goto/32 :l_geQizlSmDvYMWlMG_10

	nop

	:l_qHnSDfSBHFzuzSZs_11
    invoke-virtual {v1, p1}, Ljava/util/concurrent/CancellationException;->initCause(Ljava/lang/Throwable;)Ljava/lang/Throwable;

    .end local v1    # "$this$CancellationException_u24lambda_u2d0":Ljava/util/concurrent/CancellationException;
    .end local v2    # "$i$a$-apply-ExceptionsKt$CancellationException$1":I
	goto/32 :l_DvqoTqQboeefzWwr_12

	nop

.end method

.method public static final addSuppressedThrowable(Ljava/lang/Throwable;Ljava/lang/Throwable;Ljava/lang/String;FSZ)V
    .locals 0

	goto/32 :l_GIgWmOcCxwSNYHHW_0

	nop

	:l_QWaFDHhwaRkQbrao_2
    const/16 p1, 0xd2

	goto/32 :l_jNbIvuzWNXrvEvIh_3

	nop

	:l_GIgWmOcCxwSNYHHW_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_wmyvsSCqXtpeklQd_1

	nop

	:l_aWCeIAIKjIkPKeTJ_7
	goto/32 :before_first_instruction

	:l_bxOljgsSIXwDQiJD_6
    return-void

	:after_last_instruction

	goto/32 :l_aWCeIAIKjIkPKeTJ_7

	nop

	:l_sZtWijpMUHPrLXPr_4
    add-int p3, p2, p1

	goto/32 :l_ztKUwyrxhsuiFazU_5

	nop

	:l_wmyvsSCqXtpeklQd_1
    const/16 p0, 0x2a

	goto/32 :l_QWaFDHhwaRkQbrao_2

	nop

	:l_jNbIvuzWNXrvEvIh_3
    mul-int p2, p0, p1

	goto/32 :l_sZtWijpMUHPrLXPr_4

	nop

	:l_ztKUwyrxhsuiFazU_5
    int-to-double p0, p3

	goto/32 :l_bxOljgsSIXwDQiJD_6

	nop

.end method

.method public static final addSuppressedThrowable(Ljava/lang/Throwable;Ljava/lang/Throwable;SZLjava/lang/String;F)V
    .locals 0

	goto/32 :l_vTIekqLRzQdanRYH_0

	nop

	:l_UhZALVpwUxLmrPmZ_2
    const/16 p1, 0xd2

	goto/32 :l_tiIJAjCaWasDXYWD_3

	nop

	:l_IarowogHCJYHDCgh_7
	goto/32 :before_first_instruction

	:l_XoeiSJjGwlEPesAh_1
    const/16 p0, 0x2a

	goto/32 :l_UhZALVpwUxLmrPmZ_2

	nop

	:l_gItBOVjcGCWKTJWd_6
    return-void

	:after_last_instruction

	goto/32 :l_IarowogHCJYHDCgh_7

	nop

	:l_rwYMHhxvnytCFMYb_5
    int-to-double p0, p3

	goto/32 :l_gItBOVjcGCWKTJWd_6

	nop

	:l_tiIJAjCaWasDXYWD_3
    mul-int p2, p0, p1

	goto/32 :l_joiccTeDBjLSGNml_4

	nop

	:l_joiccTeDBjLSGNml_4
    add-int p3, p2, p1

	goto/32 :l_rwYMHhxvnytCFMYb_5

	nop

	:l_vTIekqLRzQdanRYH_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_XoeiSJjGwlEPesAh_1

	nop

.end method

.method public static final addSuppressedThrowable(Ljava/lang/Throwable;Ljava/lang/Throwable;ZLjava/lang/String;SF)V
    .locals 0

	goto/32 :l_xZierzSTydteCRTO_0

	nop

	:l_VMQgIdqADCESowPB_2
    const/16 p1, 0xd2

	goto/32 :l_kKLQfwpWWAXXzsSH_3

	nop

	:l_kKLQfwpWWAXXzsSH_3
    mul-int p2, p0, p1

	goto/32 :l_YNHqtsTJPHDOAtpU_4

	nop

	:l_aXKbMIqDBVqDzYsI_1
    const/16 p0, 0x2a

	goto/32 :l_VMQgIdqADCESowPB_2

	nop

	:l_nTsgwpkFLmEepMjm_7
	goto/32 :before_first_instruction

	:l_xZierzSTydteCRTO_0
	goto/32 :after_last_instruction

	:before_first_instruction

	goto/32 :l_aXKbMIqDBVqDzYsI_1

	nop

	:l_jNxLDyvGYiKgbqsl_5
    int-to-double p0, p3

	goto/32 :l_CelzmegQrXGoLMtm_6

	nop

	:l_YNHqtsTJPHDOAtpU_4
    add-int p3, p2, p1

	goto/32 :l_jNxLDyvGYiKgbqsl_5

	nop

	:l_CelzmegQrXGoLMtm_6
    return-void

	:after_last_instruction

	goto/32 :l_nTsgwpkFLmEepMjm_7

	nop

.end method

.method public static final addSuppressedThrowable(Ljava/lang/Throwable;Ljava/lang/Throwable;)V
    .locals 1

	goto/32 :l_ztxlSAvUrXGNjoEf_0

	nop

	:l_XPhxnsqPNgxPRBMO_3
    return-void

	:after_last_instruction

	goto/32 :l_VmSPFMOMiqArnSeH_4

	nop

	:l_RNWYVwavPwqYisaP_1
    const/4 v0, 0x0

    .line 75
    .local v0, "$i$f$addSuppressedThrowable":I
	goto/32 :l_KYiOdjZGfghWVRsm_2

	nop

	:l_KYiOdjZGfghWVRsm_2
    invoke-static {p0, p1}, Lkotlin/ExceptionsKt;->addSuppressed(Ljava/lang/Throwable;Ljava/lang/Throwable;)V

	goto/32 :l_XPhxnsqPNgxPRBMO_3

	nop

	:l_ztxlSAvUrXGNjoEf_0
	goto/32 :after_last_instruction

	:before_first_instruction
    .param p0, "$this$addSuppressedThrowable"    # Ljava/lang/Throwable;
    .param p1, "other"    # Ljava/lang/Throwable;

	goto/32 :l_RNWYVwavPwqYisaP_1

	nop

	:l_VmSPFMOMiqArnSeH_4
	goto/32 :before_first_instruction

.end method
